File: Programing_Basics_with_JS/examPreparation/programmingBasicsOnlineExam6And7April2019/06.CinemaTickets.js
function cinemaTickets(input) {
    let index = 0;
    let command = input[index];
    index++;

    let totalStudentTickets = 0;
    let totalStandardTickets = 0;
    let totalKidsTickets = 0;

    while(command !== 'Finish') {
        let movieName = command;
        let roomFreeSpace = Number(input[index]);
        index++;
        let occupancyCount = 0;
        newCommand = input[index];
        index++;

        while(newCommand !== 'End') {
            let kindOfTickets = newCommand;
            occupancyCount++;
            
            switch(kindOfTickets) {
                case 'student': totalStudentTickets++; break;
                case 'standard': totalStandardTickets++; break;
                case 'kid': totalKidsTickets++; break;
            }

            if (roomFreeSpace === occupancyCount) {
                break;
            }

            newCommand = input[index];
            index++;
        }

        let roomOccupancyPercent = occupancyCount / roomFreeSpace * 100;
        console.log(`${movieName} - ${roomOccupancyPercent.toFixed(2)}% full.`);

        command = input[index];
        index++;
    }

    let allTickets = totalStudentTickets + totalStandardTickets + totalKidsTickets;
    let studentTicketsPercent = totalStudentTickets / allTickets * 100;
    let standardTicketsPercent = totalStandardTickets / allTickets * 100;
    let kidTicketsPercent = totalKidsTickets / allTickets * 100;

    console.log(`Total tickets: ${allTickets}`);
    console.log(`${studentTicketsPercent.toFixed(2)}% student tickets.`);
    console.log(`${standardTicketsPercent.toFixed(2)}% standard tickets.`);
    console.log(`${kidTicketsPercent.toFixed(2)}% kids tickets.`);
}

cinemaTickets(["Taxi",
"10",
"standard",
"kid",
"student",
"student",
"standard",
"standard",
"End",
"Scary Movie",
"6",
"student",
"student",
"student",
"student",
"student",
"student",
"Finish"])

cinemaTickets(["The Matrix",
"20",
"student",
"standard",
"kid",
"kid",
"student",
"student",
"standard",
"student",
"End",
"The Green Mile",
"17",
"student",
"standard",
"standard",
"student",
"standard",
"student",
"End",
"Amadeus",
"3",
"standard",
"standard",
"standard",
"Finish"])

File: Programming_Basics_with_Python/programming_basics_online_exam_6_and_7_april_2019/06._cinema_tickets.py
command = input()

all_tickets_count = 0
all_student_tickets_count = 0
all_standard_tickets_count = 0
all_kid_tickets_count = 0

while not command == 'Finish':
    film_name = command
    free_room_seats = int(input())

    current_film_tickets_count = 0
    current_film_student_tickets_count = 0
    current_film_standard_tickets_count = 0
    current_film_kid_tickets_count = 0

    new_command = input()
    while not new_command == 'End':
        tickets_type = new_command
        current_film_tickets_count += 1
        all_tickets_count += 1
        if tickets_type == 'student':
            current_film_student_tickets_count += 1
            all_student_tickets_count += 1
        elif tickets_type == 'standard':
            current_film_standard_tickets_count += 1
            all_standard_tickets_count += 1
        elif tickets_type == 'kid':
            current_film_kid_tickets_count += 1
            all_kid_tickets_count += 1

        if current_film_tickets_count == free_room_seats:
            break
        else:
            new_command = input()

    sold_tickets_percent = current_film_tickets_count / free_room_seats * 100
    print(f'{film_name} - {sold_tickets_percent:.2f}% full.')

    command = input()

student_tickets_percent = all_student_tickets_count / all_tickets_count * 100
standard_tickets_percent = all_standard_tickets_count / all_tickets_count * 100
kid_tickets_percent = all_kid_tickets_count / all_tickets_count * 100

print(f'Total tickets: {all_tickets_count}')
print(f'{student_tickets_percent:.2f}% student tickets.')
print(f'{standard_tickets_percent:.2f}% standard tickets.')
print(f'{kid_tickets_percent:.2f}% kids tickets.')